.github/workflows/unit_and_integration_tests.yml

```diff
@@ -0,0 +1,31 @@
+name: Unit & Integration Tests
+
+on: [push, pull_request]
+
+jobs:
+  test:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [2.7, 3.7, 3.9]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox
+    - name: Run Tox
+      run: tox -e py
+    - uses: actions/upload-artifact@v2
+      if: failure()
+      with:
+        name: ${{ matrix.os }}_py${{ matrix.python-version }}_log.html
+        path: .tox/py_log.html

```

Comment:
looks good.

Does rototframework support non HTML report? Like simple text?
If it does, you can just show the file in the log... in this way, the log easy to see in the same build logs... and you don't have to go to extra artifact links..

Or maybe use `pip install html2text` and then run html2text on the HTML log file. It will show markdown, but is good enough as text.

Reply:
There is a simple textual output printed to stdout by default. This is already viewable in the job output log (see screenshot). This will also display the top level error for each test case (if any). In a lot of cases this will be enough to diagnose the issue and push a fix, but the log file is useful to have available if the issue is a bit more obscure. It's possible to do some post processing on the output.xml to extract specific data but I think at this point there's no harm in uploading the log file. Its the best artifact to use when diagnosing issues, they're reasonably small in size with a retention period and there is little to no impact on the build time.


![image](https://user-images.githubusercontent.com/1598192/103438234-03706600-4c29-11eb-83f4-35e309ef3515.png)